cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -4927,6 +4927,160 @@ fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
                 }
             }
         }
+        Opcode::IaddPairwise => {
+            if let Some(swiden_low) = matches_input(ctx, inputs[0], Opcode::SwidenLow) {
```

Comment:
These nested conditionals could be written together in one level, I think; this would also avoid the subtle asymmetry (possibly leading to a bug with later additions) that the `else if` attaches only to the outer `if`. Something like:

`if let (Some(swiden_low), Some(swiden_high)) = (matches_input(...), matches_input(...)) {`